pkg/controller/association/controller/es_monitoring.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package controller
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	esv1 "github.com/elastic/cloud-on-k8s/pkg/apis/elasticsearch/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/association"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/operator"
+	eslabel "github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/label"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/elasticsearch/user"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/rbac"
+)
+
+const (
+	// EsMonitoringAssociationLabelName marks resources created by this controller for easier retrieval.
+	EsMonitoringAssociationLabelName = "esmonitoringassociation.k8s.elastic.co/name"
+	// EsMonitoringAssociationLabelNamespace marks resources created by this controller for easier retrieval.
+	EsMonitoringAssociationLabelNamespace = "esmonitoringassociation.k8s.elastic.co/namespace"
+	// EsMonitoringAssociationLabelType marks the type of association.
+	EsMonitoringAssociationLabelType = "esmonitoringassociation.k8s.elastic.co/type"
```

Comment:
I think the labelling logic of other association controllers is slightly different, and closer to something like:

```
// to which ES cluster is this association coming from
esassociation.k8s.elastic.co/name: es-name
esassociation.k8s.elastic.co/namespace: default
// what's the type of association here?
esassociation.k8s.elastic.co/type: es-monitoring  // associated to ES, for monitoring reasons (in case we add other association reasons later on)
```

so the label name means "this is an association to elasticsearch", and the label values mean "this is the associated elasticsearch, and the type of association for that elasticsearch".

Here it's slightly different since the type of association is reflected in the label name? WDYT? (def not a big deal, trying to be as consistent as possible)

Reply:
Makes sense.